src/test/transaction_tests.cpp

```diff
@@ -882,13 +884,13 @@ BOOST_AUTO_TEST_CASE(test_IsStandard)
 
     // Only one TxoutType::NULL_DATA permitted in all cases
     t.vout.resize(2);
-    t.vout[0].scriptPubKey = CScript() << OP_RETURN << ParseHex("04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38");
+    t.vout[0].scriptPubKey = CScript() << OP_RETURN << Vec(HexLiteral<uint8_t>("04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38"));
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/30377#discussion_r1720759053

> nit: I'm not in love with the `ParseHex` -> `Vec(HexLiteral<uint8_t>` change, seems more noisy than needed. What's the practical problem with keeping `ParseHex` in the tests?

I feel like we could add a `HexLiteralU` or similar convenience function that is equivalent to `HexLiteral<uint8_t>`, because `HexLiteral<uint8_t>` does seem to be needed a lot of places after the `std::byte` change (107 instances currently, which is more than I would have expected).

It would also be nice to have a followup adding std::array support to CScript so more Vec calls can be dropped, too, but we should avoid touching CScript implementation in this PR. It looks like there are 28 Vec(HexLiteral()) calls currently, and if std::array support is added to CScript 15 of them can be dropped.

I think suggestion to keep ParseHex in tests would be a step backward because compile time checking of strings in tests is as much (or more) of a practical benefit there as in other parts of the code. I think most of the practical benefit is just knowing code works without having to wait for tests to run.

Reply:
Thanks @ryanofsky